controller/main.go

```diff
@@ -87,14 +87,18 @@ func (c *controller) SetBalancer(l log.Logger, name string, svcRo *v1.Service, _
 	}
 
 	if !reflect.DeepEqual(svcRo.Status, svc.Status) {
-		var st v1.ServiceStatus
-		st, svc = svc.Status, svcRo.DeepCopy()
-		svc.Status = st
-		if err := c.client.UpdateStatus(svc); err != nil {
-			level.Error(l).Log("op", "updateServiceStatus", "error", err, "msg", "failed to update service status")
-			return controllers.SyncStateError
-		}
+		updateSvcStatus(svc, svcRo)
+	}
+
+	if !reflect.DeepEqual(svcRo.Annotations, svc.Annotations) {
+		updateSvcAnnotations(svc, svcRo)
 	}
+
+	if err := c.client.UpdateStatus(svc); err != nil {
```

Comment:
You need to call this only if one of the deepequals above changed. I suggest using a flag.

Reply:
sorry I don't notice this

Reply:
I realize only now that this might not work because metallb has writing permissions only on the status subresource https://github.com/metallb/metallb/blob/main/config/rbac/role.yaml#L19 
So _it might be_ that this requires additional permissions, and if that is the case I am not sure it's worth it. If it's needed, we can pivot to expose this information via one of the status crds we are working to expose as part of https://github.com/metallb/metallb/pull/1685

Reply:
Nope, I made an image with the current branch and did a test for it, and the annotaitons of the service were successfully updated.